include/blas/counter.hh

```diff
@@ -709,15 +732,19 @@ private:
     //--------------------------------------------------------------------------
     /// Constructor initializes PAPI counting set on first call to get().
     counter():
-        set_( nullptr )
+        set_( nullptr ),
+        total_flop_count_(-1)
     {
         #ifdef BLAS_HAVE_PAPI
             papi_sde::PapiSde sde( "blas" );
             set_ = sde.create_counting_set( "counter" );
+            total_flop_count_ = 0;
+            sde.register_counter_cb("flops", PAPI_SDE_RO|PAPI_SDE_DELTA, get_flop_count, total_flop_count_);
         #endif
     }
 
```

Comment:
The `gpu_intel` build complains about `set_` being unused. I wonder if it should be `#ifdef BLAS_HAVE_PAPI`? Which raises question if `total_flop_count_` should also be inside `#ifdef`.
```
./include/blas/counter.hh:746:18: error: private field 'set_' is not used [-Werror,-Wunused-private-field]
    CountingSet* set_;
                 ^
1 error generated.
```
(We use `-Werror` in our checks, to require no compiler warnings. I'm pretty sure the `gpu_intel` build uses the Intel icpx compiler instead of g++.)

Reply:
I moved the initialization inside the constructor. This resolved the problem for a small unit-test that reproduced the problem, as tested on methane with icpx (Intel(R) oneAPI DPC++/C++ Compiler 2023.2.0 (2023.2.0.20230721))